### PR TITLE
Workaround to support ExPlat in Jetpack app

### PR DESCRIPTION
**Pull request by salimbraksa**
Ref p1682352977248619-slack-C04M1NX8YCR

## Description

This PR does the following:
- Updates `Automattic-Tracks-iOS` library from `0.13` to `2.0.0+`.
- Tweak ExPlat configuration to make it work for Jetpack. But there are no experiments to test in this PR yet.

## Related PRs

**This PR is blocked By:**
- https://github.com/Automattic/Automattic-Tracks-iOS/pull/253

**And blocking:**
- https://github.com/wordpress-mobile/WordPress-iOS/pull/20590

## Test Instructions

>Updates `Automattic-Tracks-iOS` library from `0.13` to `2.0.0+`.

Because of the change, we should double check Track events are working as expected.

1. Set a breakpoint in `TracksService.m` line 141.
```objective-c
- (void)trackEventName:(NSString *)eventName withCustomProperties:(NSDictionary *)customProperties
{
    eventName = [NSString stringWithFormat:@"%@_%@", self.eventNamePrefix, eventName];
    
🔵  [self.tracksEventService createTracksEventWithName:eventName
                                              username:self.username
                                                userID:self.userID
```
2. Build and run the Jetpack / WordPress app.
3. Navigate to any screen until the app is paused.
4. Head to the debugger and type `po eventName`.
5. **Expect** the event name to be logged and starts with `jpios` or `wpios` depending on whether the running app is Jetpack or WordPress.

## Regression Notes
1. Potential unintended areas of impact
`Automattic-Tracks-iOS` pod was updated from `0.13` to `2.0.0`, which is a significant update. We should smoke test whether track events are working as expected. See Test Instructions.

6. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual Testing.

7. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] VoiceOver.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] iPhone and iPad. 
- [x] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Review by mkevins — APPROVED** [review]
This tested well for me - though the prefixes (`jpios` and `wpios`) were seemingly truncated in the logs after `po eventName`, they were easily visible when I moused over the `eventName` variable in the code view.

File: Podfile [hunk 1]
```diff
@@ -404,7 +404,9 @@ pre_install do |installer|
   installer.pod_targets.each do |pod|
     # Statically linking Sentry results in a conflict with `NSDictionary.objectAtKeyPath`, but dynamically
     # linking it resolves this.
-    if pod.name == 'Sentry'
+    #
+    # The regex is to match both "Sentry" and "SentryPrivate" pods.
+    if pod.name =~ /^Sentry/
```
Comment (salimbraksa) [inline Podfile]: @mokagio @crazytonyli Let me know if this change is okey. I couldn't update Sentry to 8.5.0 without this change. 

Apparently, Sentry 8.5.0 comes with another pod called `SentryPrivate` and it needs to be installed dynamically alongside the `Sentry` pod. Thus, the change.

Comment (crazytonyli) [reply Podfile]: Nitpick: it's probably best to list out the specific pods if there are only two pods? i.e. `if %w[Sentry SentryPrivate].include? pod.name`?

File: WordPress/Jetpack/AppConstants.swift [hunk 2]
```diff
@@ -14,6 +14,7 @@ import WordPressKit
     static let shareAppName: ShareAppName = .jetpack
     static let mobileAnnounceAppId = "6"
     @objc static let eventNamePrefix = "jpios"
+    @objc static let explatPlatform = "wpios"
```
Comment (mokagio) [inline WordPress/Jetpack/AppConstants.swift:17]: Should this still be `wpios` even though it's in the Jetpack file?

**Comment by mokagio** [comment]
I'm going to start the code freeze shortly. I'm unsure whether these changes are required to ship with 22.3 or can wait till the next version. As such, I won't take over merging but I'll also leave the milestone as [22.3](https://github.com/wordpress-mobile/WordPress-iOS/milestone/245) so we can followup with a release branch merge if needed.

File: WordPress/Jetpack/AppConstants.swift (hunk 2 above)
Comment (salimbraksa) [reply WordPress/Jetpack/AppConstants.swift:17]: That's intentional; it's the workaround to make ExPlat work in Jetpack. See p1682352977248619-slack-C04M1NX8YCR

File: Podfile (hunk 1 above)
Comment (salimbraksa) [reply Podfile]: @crazytonyli Resolved in 450c777cf959d0d1e2d4b030c5eb4fa51281cf14

File: WordPress/Jetpack/AppConstants.swift (hunk 2 above)
Comment (mokagio) [reply WordPress/Jetpack/AppConstants.swift:17]: Perfect, thank you for confirming @salimbraksa

**Comment by mokagio** [comment]
@salimbraksa I published a new version of Tracks with your changes from https://github.com/Automattic/Automattic-Tracks-iOS/pull/253. I was going to merge this PR and ship a new beta, then I noticed https://github.com/wordpress-mobile/WordPress-iOS/pull/20613. I won't be shipping a beta till that one is merged, too.

**Comment by mkevins** [comment]
Thanks Gio! #20613 is now merged. :+1:

**Pull request merged**